src/main/java/org/mule/extension/api/metadata/NullMetadataKey.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.api.metadata;
+
+
+import org.mule.api.metadata.MetadataKey;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * {@link MetadataKey} implementation that represents the absence of a Key
+ */
+public final class NullMetadataKey implements MetadataKey
```

Comment:
When an Operation has no MetadataKeyParameter but the @Content or Output do have dynamic types. Then, a NullMetadataKey is passed on to the getContent and getOutput metadata instead of having multiple signatures or just passing "null".

This covers the current "MetadataStaticParam" case